quipucords/api/deployments_report/model.py

```diff
@@ -140,6 +153,37 @@ def cached_csv(self, data):
             f.write(data)
         self.cached_csv_file_path = file_path
 
+    def _rerun_latest_fingerprint(self, wait=False) -> None:
+        """
+        Trigger the latest fingerprint task to rerun.
+
+        Why would you want to do this? If you expect the cached_fingerprints or
+        cached_csv data to exist but either is missing, you may recreate them by
+        calling this method.
+
+        :param wait: if true, wait for task to complete
+        """
+        logger.info("Rerunning the last fingerprint task for %s", self)
+
+        # local imports to avoid possible import loop
+        from api.scantask.model import ScanTask
+        from scanner import tasks
+
+        scan_task = (
+            ScanTask.objects.filter(
+                job__report__deployment_report_id=self.id,
+                scan_type=ScanTask.SCAN_TYPE_FINGERPRINT,
+            )
+            .order_by("-id")
+            .first()
+        )
+        logger.info("Found %s for %s", scan_task, self)
+        if not scan_task:
+            raise ScanTask.DoesNotExist(f"No fingerprint ScanTask is related to {self}")
+        fingerprint_task = tasks.fingerprint.delay(scan_task_id=scan_task.id)
+        if wait:
+            fingerprint_task.get()
```

Comment:
What if instead of invoking celery when `wait` is true we just run the function synchronously, w/o celery?

Reply:
Unfortunately, we can't invoke `tasks.fingerprint` directly because we defined it as a task with `bind=True` which means it _requires_ a `self` argument pointing to a `celery.Task` which we don't have when calling it synchronously. That `self` is passed into lower layers that check if the `ScanJob` has been cancelled/revoked. Even though that check is not relevant to us in this context, it is not trivial to refactor it out.